File: samples/conference/management-bc/src/main/java/com/microsoft/conference/management/domain/Events/SeatsReservationCancelled.cs
using System;
using System.Collections.Generic;
using ENode.Eventing;

namespace ConferenceManagement
{
    public class SeatsReservationCancelled : DomainEvent<Guid>
    {
        public Guid ReservationId { get; set; }
        public IEnumerable<SeatAvailableQuantity> SeatAvailableQuantities { get; set; }

        public SeatsReservationCancelled() { }
        public SeatsReservationCancelled(Guid reservationId, IEnumerable<SeatAvailableQuantity> seatAvailableQuantities)
        {
            ReservationId = reservationId;
            SeatAvailableQuantities = seatAvailableQuantities;
        }
    }
}

File: samples/conference/management-bc/src/main/java/com/microsoft/conference/management/domain/Events/SeatsReservationCancelled.java
package com.microsoft.conference.management.domain.Events;

import com.enodeframework.eventing.DomainEvent;
import com.microsoft.conference.management.domain.Models.SeatAvailableQuantity;

import java.util.List;

public class SeatsReservationCancelled extends DomainEvent<String> {
    public String ReservationId;
    public List<SeatAvailableQuantity> SeatAvailableQuantities;

    public SeatsReservationCancelled() {
    }

    public SeatsReservationCancelled(String reservationId, List<SeatAvailableQuantity> seatAvailableQuantities) {
        ReservationId = reservationId;
        SeatAvailableQuantities = seatAvailableQuantities;
    }
}
